File: Generated_test_cases/JS/src/pipeline.js
const { execSync } = require('child_process');

function sanitize(cmd) {
    // VULN-5: Incomplete sanitization – only removes ; | &
    return cmd.replace(/[;|&]/g, '');
}

function execute(cmd, input) {
    const safeCmd = sanitize(cmd);
    if (!safeCmd) return '';
    try {
        return execSync(safeCmd, { input, encoding: 'utf8' });
    } catch (e) {
        return '';
    }
}

module.exports = { execute };

File: Generated_test_cases/GO/pipeline/pipeline.go
package pipeline

import (
	"os/exec"
	"strings"
)

// VULN-5: Command injection – incomplete sanitizer
func Execute(cmd, input string) string {
	safe := sanitize(cmd)
	if safe == "" {
		return ""
	}
	c := exec.Command("/bin/sh", "-c", safe)
	out, _ := c.CombinedOutput()
	return string(out)
}

func sanitize(cmd string) string {
	// removes only ; | &
	r := strings.NewReplacer(";", "", "|", "", "&", "")
	return r.Replace(cmd)
}